docs/source/plugins/for_plugin_developers.rst

```diff
@@ -0,0 +1,291 @@
+.. _plugins-for-plugin-developers:
+
+.. role:: python(code)
+   :language: python
+
+creating a napari plugin
+========================
+
+This document explains how to extend napari functionality by writing a plugin
+that can be installed with ``pip`` and autodetected by napari.  For in-depth
+information on how plugins are used internally in napari, see
+:ref:`plugins-for-napari-developers`.
+
+
+Overview
```

Comment:
Do you want to mention the existence of the cookiecutter up here with a link below?